front/lib/actions/mcp_internal_actions/servers/http_client.ts

```diff
@@ -0,0 +1,282 @@
+      },
+      async ({ url, method, headers, body, timeout_ms }) => {
+        const timeoutMs = timeout_ms ?? DEFAULT_TIMEOUT_MS;
+        const controller = new AbortController();
+        const timeoutId = setTimeout(() => controller.abort(), timeoutMs);
+
+        const requestHeaders = validateAndSanitizeHeaders(headers);
+
+        const bearerToken = await getBearerToken(auth, agentLoopContext);
+        if (bearerToken) {
+          requestHeaders["Authorization"] = `Bearer ${bearerToken}`;
+        }
+
+        try {
+          const response = await untrustedFetch(url, {
+            method,
+            headers: requestHeaders,
+            body: body ?? undefined,
+            signal: controller.signal,
+          });
+
+          clearTimeout(timeoutId);
+
+          let responseBody = "";
+          const contentType =
+            response.headers.get("content-type")?.toLowerCase() ?? "";
+
+          if (
+            TEXT_BASED_CONTENT_TYPES.some((type) => contentType.includes(type))
+          ) {
+            const bodyText = await response.text();
+
+            // Limit response size to prevent overwhelming the context
+            if (bodyText.length > MAX_RESPONSE_SIZE) {
+              responseBody =
+                bodyText.slice(0, MAX_RESPONSE_SIZE) +
+                `\n\n[... response truncated, exceeded ${MAX_RESPONSE_SIZE} bytes]`;
+            } else {
+              responseBody = bodyText;
+            }
+          } else {
+            responseBody = `[Content type not displayed. Content-Type: ${contentType}, Size: ${response.headers.get("content-length") ?? "unknown"} bytes]`;
+          }
+
+          const resultText = [
+            `HTTP ${response.status} ${response.statusText}`,
+            "",
+            "Response Headers:",
+            ...Object.entries(response.headers)
+              .filter(([key]) =>
+                SAFE_RESPONSE_HEADERS.includes(key.toLowerCase())
+              )
+              .map(([key, value]) => `  ${key}: ${value}`),
+            "",
+            "Response Body:",
+            responseBody,
+          ].join("\n");
+
+          return new Ok([
+            {
+              type: "text",
+              text: resultText,
+            },
+          ]);
+        } catch (error) {
+          clearTimeout(timeoutId);
+
+          if (error instanceof Error) {
+            if (error.name === "AbortError") {
+              return new Err(
+                new MCPError(`Request timed out after ${timeoutMs}ms`)
+              );
+            }
+            return new Err(
+              new MCPError(
+                `HTTP request failed: ${normalizeError(error).message}`
+              )
+            );
+          }
+
+          return new Err(
+            new MCPError(
+              `HTTP request failed: ${normalizeError(error).message}`
+            )
+          );
+        }
+      }
+    )
+  );
+
+  registerWebSearchTool(auth, server, agentLoopContext);
```

Comment:
@frankaloia ?? why this ooc? This is very weird for the http client server to overlap with the browsing server?

Reply:
I think this is too magical. Let's document adding web browsing as a strong suggestion but if we add shortcut like that everywhere we:

- won't have any clue anymore what does what
- will have repeated tools all over the place

Please remove and document 🙏